src/python/Utils/TokenManager.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+#pylint: disable=W0212
+"""
+File       : TokenManager.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: IAM token manager
+"""
+
+# system modules
+import os
+import ssl
+import time
+import logging
+import traceback
+
+# third part library
+try:
+    import jwt
+except ImportError:
+    traceback.print_exc()
+    jwt = None
+
+# prevent "SSL: CERTIFICATE_VERIFY_FAILED" error
+# this will cause pylint warning W0212, therefore we ignore it above
+ssl._create_default_https_context = ssl._create_unverified_context
+
+
+def getToken(name=None):
+    """
+    Get IAM token either from environment or file name
+    :param name: ether file name containing token or environment name which hold the token value.
+    If not provided it will be assumed to read token from IAM_TOKEN environment.
+    :return: token or None
+    """
+    if name and os.path.exists(name):
+        token = None
+        with open(name) as istream:
+            token = istream.read()
+        return token
+    if name:
+        return os.environ.get(name)
+    return os.environ.get("IAM_TOKEN")
+
+def tokenData(token, url=None, aud=None):
+    """
+    inspect and extract token data
+    :param token: token string
+    :param url: IAM provider URL, by default it is https://cms-auth.web.cern.ch/jwk
+    :param aud: audience string, by default it is https://wlcg.cern.ch/jwt/v1/any
+    """
+    if not token or not jwt:
+        return {}
+    if isinstance(token, str):
+        token = bytes(token, 'utf-8')
+    if not url: # use CMS IAM
+        url = "https://cms-auth.web.cern.ch/jwk"
+    jwksClient = jwt.PyJWKClient(url)
+    signingKey = jwksClient.get_signing_key_from_jwt(token)
+    key = signingKey.key
+    headers = jwt.get_unverified_header(token)
+    alg = headers.get('alg', 'RS256')
+    if not aud: # use WLCG
+        aud = "https://wlcg.cern.ch/jwt/v1/any"
+    data = jwt.decode(
+        token,
+        key,
+        algorithms=[alg],
+        audience=aud,
+        options={"verify_exp": True},
+    )
+    return data
+
+def validToken(token):
```

Comment:
Perhaps calling it `isTokenValid` would be better?

Somehow unrelated to this, but we should also have a function/method that returns the lifetime of a given token (such that we can get a new one *before* we see it expired)

Reply:
ok